sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,472 @@
+
+    >>> from sympy.physics.continuum_mechanics.beam import Beam
+    >>> from sympy import Symbol, Piecewise
+    >>> x = Symbol('x')
+    >>> E = Symbol('E')
+    >>> I = Symbol('I')
+    >>> b = Beam(4, E, I)
+    >>> b.apply_load(value=-9, start=4, order=-1)
+    >>> b.apply_load(value=-3, start=0, order=-1)
+    >>> b.apply_load(order=0, start=2, value=6)
+    >>> b.bc_deflection = [(4, 0)]
+    >>> b.boundary_conditions
+    {'deflection': [(4, 0)], 'moment': [], 'slope': []}
+    >>> b.load
+    -3*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*SingularityFunction(x, 4, -1)
+    >>> b.shear_force()
+    -3*SingularityFunction(x, 0, 0) + 6*SingularityFunction(x, 2, 1) - 9*SingularityFunction(x, 4, 0)
+    >>> b.bending_moment()
+    3*SingularityFunction(x, 0, 1) - 3*SingularityFunction(x, 2, 2) + 9*SingularityFunction(x, 4, 1)
+    >>> b.slope()
+    (3*SingularityFunction(x, 0, 2)/2 - SingularityFunction(x, 2, 3) + 9*SingularityFunction(x, 4, 2)/2 - 7)/(E*I)
+    >>> b.deflection()
+    (-7*x + SingularityFunction(x, 0, 3)/2 - SingularityFunction(x, 2, 4)/4 + 3*SingularityFunction(x, 4, 3)/2)/(E*I)
+    >>> b.deflection().rewrite(Piecewise)
+    (-7*x + Piecewise((x**3, x > 0), (0, True))/2
+          + 3*Piecewise(((x - 4)**3, x - 4 > 0), (0, True))/2
+          - Piecewise(((x - 2)**4, x - 2 > 0), (0, True))/4)/(E*I)
+    """
+
+    def __init__(self, length, elastic_modulus, second_moment, variable=Symbol('x')):
+        """Initializes the class.
+
+        Parameters
+        ==========
+        length :
+            A Symbol or value representing the Beam's length.
+        elastic_modulus : Sympifyable
+            A SymPy expression representing the Beam's Modulus of Elasticity.
+            It is a measure of the stiffness of the Beam material.
+        second_moment : Sympifyable
+            A SymPy expression representing the Beam's Second moment of area.
+            It is a geometrical property of an area which reflects how its
+            points are distributed with respect to its neutral axis.
+        variable : Symbol
+            A Symbol object that will be used as the variable along the beam
+            while representing the load, shear, moment, slope and deflection
+            curve. By default, it is set to ``Symbol('x')``.
+        """
+        self._length = sympify(length)
+        self._elastic_modulus = sympify(elastic_modulus)
+        self._second_moment = sympify(second_moment)
+        if isinstance(variable, Symbol):
+            self._variable = variable
+        else:
+            raise TypeError("""The variable should be a Symbol object.""")
+        self._boundary_conditions = {'deflection': [], 'moment': [], 'slope': []}
+        self._load = 0
+
+    def __str__(self):
+        str_sol = 'Beam(%s, %s, %s)' % (sstr(self._length), sstr(self._elastic_modulus), sstr(self._second_moment))
+        return str_sol
+
+    __repr__ = __str__
```

Comment:
The `__repr__` should give a representation that fully creates the object if copied and pasted into an interpreter. I don't think you should set this as such unless that is possible.